src/font.rs

```diff
@@ -0,0 +1,165 @@
+use crate::stream::Stream;
+use crate::Error;
+use std::fmt::Display;
+
+pub fn parse_font_families(text: &str) -> Result<Vec<FontFamily>, Error> {
+    let mut s = Stream::from(text);
+    let font_families = s.parse_font_families()?;
+
+    s.skip_spaces();
+    if !s.at_end() {
+        return Err(Error::UnexpectedData(s.calc_char_pos()));
+    }
+
+    Ok(font_families)
+}
+
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+pub enum FontFamily {
```

Comment:
I think we should return just a string at this point. Mainly because CSS4 already includes `system-ui`, `ui-serif` and [others](https://drafts.csswg.org/css-fonts/#generic-font-families), but resvg/usvg doesn't. Just a string would be more generic.

Reply:
So remove the `FontFamily` enum and just return a string? I mean I can change it, but I feel like providing this separation at this level makes it a bit easier to use it. Because if we just return a string, how would one distinguish whether "serif" is the keyword serif or actually a font named serif? By preserving the quotes for names strings? But then, we will have to implement the "unquoting" on the `resvg` side anyway, so I'm not sure why it's not better to provide that abstraction on the `svgtypes` level.

But if you want I'll change it.

Reply:
Wait, so `serif` != `"serif"`? Well, in that can we should leave the enum.

Reply:
Yes, exactly.

> The following generic family keywords are defined: ‘[serif](https://www.w3.org/TR/2018/REC-css-fonts-3-20180920/#serif)’, ‘[sans-serif](https://www.w3.org/TR/2018/REC-css-fonts-3-20180920/#sans-serif)’, ‘[cursive](https://www.w3.org/TR/2018/REC-css-fonts-3-20180920/#cursive)’, ‘[fantasy](https://www.w3.org/TR/2018/REC-css-fonts-3-20180920/#fantasy)’, and ‘[monospace](https://www.w3.org/TR/2018/REC-css-fonts-3-20180920/#monospace)’. These keywords can be used as a general fallback mechanism when an author's desired font choices are not available. As keywords, they must not be quoted.

> Font family names that happen to be the same as keyword value (‘inherit’, ‘[serif](https://www.w3.org/TR/2018/REC-css-fonts-3-20180920/#serif)’, etc.) must be quoted to prevent confusion with the keywords with the same names.